### ISSUE TITLE
lint под linux не работает, т.к. необходимо указывать команду mono oscript 

### ISSUE THREAD

**Issue by nixel2007** [xDrivenDevelopment/1c-syntax#102]
@pumbaEO commented on [Wed Mar 02 2016](https://github.com/xDrivenDevelopment/vsc-language-1c-bsl/issues/12)

Под linux комманда oscript не отрабатывает, т.к. необходимо указывать в строке запуска mono oscript.exe 

---

@nixel2007 commented on [Wed Mar 02 2016](https://github.com/xDrivenDevelopment/vsc-language-1c-bsl/issues/12#issuecomment-191406010)

Под линукс разве нет нативного бинарника? 
@EvilBeaver 

---

@pumbaEO commented on [Wed Mar 02 2016](https://github.com/xDrivenDevelopment/vsc-language-1c-bsl/issues/12#issuecomment-191407212)

Есть, там чуть сложнее надо или же делать свой скрипт запускалку, но т.к. у Андрея сборка делается под виндой и архив zip то там права на запуск не устанавливаются, что-бы нативно в linux поддерживать. 
Поэтому я чуть доработал запуск, т.е. когда вызывается процесс, то запускается mono и в параметрах передается путь к oscript

**Comment by nixel2007** [comment]
@pumbaEO @EvilBeaver поясните, пожалуйста, ситуацию с запуском oscript под линукс.

Женя говорит, что нужен запуск через `mono` да еще и exe'шника.
Это стандартное поведение для oscript на линукс системах?

Есть ли в итоге линуксовый бинарник в пакете? Если есть, то можно ли его запустить без mono?

**Comment by allustin** [comment]
@pumbaEO тебя не спасёт вот такое 

https://github.com/silverbulleters/oscript-docker/blob/master/Dockerfile

Я делаю хитрую обертку над mono oscript

**Comment by allustin** [comment]
Фишка в том чтобы сделать обертку над командной строкой `mono oscript.exe`

```
sudo echo "#!/bin/sh" > /usr/bin/oscript \
  && echo '' >> /usr/bin/oscript \
  && echo 'mono /usr/lib/osc-engine/bin/oscript.exe "$@"' >> /usr/bin/oscript \
  && chmod +x /usr/bin/oscript
```

после этого в PATH как бы возникает интерпретатор `oscript`

хуже того - так как у меня docker контейнер - я делаю alias на хост систему

```
alias oscript="docker run -it --rm -v $(pwd):/scripts -w /scripts silverbulleters/vanessa-oscript"
```

после этого уже на хост системе есть `oscript` и никто не догадывается что оно в контейнере

**Comment by pumbaEO** [comment]
В чистом виде, запуск нужно проводить с командой mono путькExe, можно создать bash файл, который будет просто все параметры передавать в эту строку запуска
cat /usr/local/bin/oscript
```
#!/bin/bash 
mono /opt/oscript/bin/oscript.exe $
``` 
chmod +x /usr/local/bin/oscript  тогда будет работать простым указанием команды "oscript", автоматом это решается пока только в случаи использования rpm/deb пакетов или ручного создания файла. 
Второй вариант это в настройках указывать путь к oscript.exe и тогда дополнительно указывать преффикс mono. 

docker - тоже половинчатое решение, т.к. будет работать без доп. настроек только на deb-based дистрибутивах, на rpm (centos, fedora, redhat) необходимо добавлять selinux контекст к пробрасываем папкам.

**Comment by nixel2007** [comment]
Если честно, мне больше нравится путь с настройкой окружения через bash-файл в local/bin. Конечному пользователю это помимо линтера в редакторах даст возможность использовать `oscript` в консоли без префиксов в виде mono.
Вариант с настройкой внутри редактора... Нравится меньше, чем bash, если честно.

Как насчет принять первый вариант за основной и описать в документации, что 
> oscript должен быть доступен в переменной окружения PATH.
Для windows-систем oscript добавляется PATH при стандартной установке. 
Для linux-систем это можно сделать вот так:
```bash
скрипт
```

?

**Comment by allustin** [comment]
@nixel2007 я так понимаю - это в принципе намек на нативную сборку oscript для linux. Чтобы было без mono.

**Comment by nixel2007** [comment]
> @nixel2007 я так понимаю - это в принципе намек на нативную сборку oscript для linux. Чтобы было без mono.

нативная сборка - это вообще идеально.
Просто не хочется смешивать ответственность редактора, окружения и интерпретатора.

**Comment by allustin** [comment]
@nixel2007 я обычно не закрываю таким задачи и держу их открытыми до "прихода Кайроса". 

[Кайрос](https://ru.wikipedia.org/wiki/%D0%9A%D0%B0%D0%B9%D1%80%D0%BE%D1%81) придет вместе с нативной сборкой. А пока ты скорее всего прав - поможет описание в документации.

Есть вот такое 

https://gist.github.com/allustin/f9718b1c7f02a50d85fd

и замечания @EvilBeaver 
https://gist.github.com/EvilBeaver/fee28f5a18ccc66855c6

Последний финально рабочий.
Ну и docker я использован его аналог

**Comment by EvilBeaver** [comment]
@allustin ты бы уже смержил чтоли мой gist в свой, я там даже скрипт в каментах написал.

**Comment by EvilBeaver** [comment]
Да, и расскажите мне, что такое "нативная сборка под linux, чтобы без mono". Это как?

**Comment by awa15** [comment]
@EvilBeaver , переписать на c++, не?

**Comment by pumbaEO** [comment]
Давайте подведем черту: 
в wiki добавляем пример bash скрипта для прозачного запуска oscript. 
В настройках я бы все-таки оставил вомзожность указания произвольного пути - данная фича пригодится для различных тестов новых релизов, когда у тебя в path прописан stable, а в настройках проекта указал другой путь (тогда нет необходимости играться с запуском vscode из под определенного bat файла). 
Из текущей реализации убрать переопределение в строке запуска mono в случаии если linux.

**Comment by nixel2007** [comment]
> @EvilBeaver , переписать на c++, не?

точно, оно же на шарпе...

**Comment by EvilBeaver** [comment]
@awa15 по-хорошему, надо бы. Но я плюсы знаю совсем плохо.

**Comment by artbear** [comment]
Интересная мысль по С++. Я бы в плюсы с удовольствием вернулся :)

чт, 3 мар. 2016 г. в 23:37, Andrei Ovsiankin <notifications@github.com>:

**Comment by allustin** [comment]
@nixel2007 @awa15 @EvilBeaver 

почему на плюсы... мы же все уже знаем *на чем* сейчас правильно писать свои интерпретаторы ;-)

**Comment by EvilBeaver** [comment]
Я догадываюсь, что ты про go, но при чем здесь именно интерпретаторы?

**Comment by pumbaEO** [comment]
Java.
Холивар к задаче не имеющий отношения.
04 марта 2016 г. 7:43 пользователь "Andrei Ovsiankin" <
notifications@github.com> написал:

**Comment by nixel2007** [comment]
Информация добавлена в wiki пакетов

**Comment by theshadowco** [comment]
Наверное стоит закрыть?
